lib/api/controllers/adminController.js

```diff
@@ -102,6 +102,49 @@ function AdminController(kuzzle) {
   };
 
   /**
+   * Returns the Kuzzle configuration
+   *
+   * @param {RequestObject} requestObject
+   * @returns {Promise} promise resolved as a ResponseObject
+   */
+  this.getConfig = requestObject => {
+    var modifiedRequestObject = null;
+
+    return kuzzle.pluginsManager.trigger('data:beforeGetConfig', requestObject)
+      .then(newRequestObject => {
+        var pluginsConfig = {};
+
+        modifiedRequestObject = newRequestObject;
+
+        return kuzzle.internalEngine
```

Comment:
You have exactly what you expected in `kuzzle.pluginsManager.plugins`.

You could simply do: 
```javascript
this.getConfig = requestObject => {
  var modifiedRequestObject = null;

  return kuzzle.pluginsManager.trigger('data:beforeGetConfig', requestObject)
    .then(newRequestObject => {
      var pluginsConfig = {};

      modifiedRequestObject = newRequestObject;

      return Promise.resolve({
          kuzzle: kuzzle.config,
          plugins: kuzzle.pluginsManager.plugins
        });
     })
    .then(response => kuzzle.pluginsManager.trigger('data:afterGetConfig', new ResponseObject(modifiedRequestObject, response)));
  };
```